### PR TITLE
[cli] Add view function support

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds view function support to the CLI, tested with adding to hello blockchain

### Test Plan

```
aptos move view --function-id devnet::message::get_message --profile devnet --args address:devnet
{
  "Result": [
    "ViewFunctions!"
  ]
}

aptos move run  --function-id devnet::message::set_message --profile devnet --args "string:I can use view functions"
Do you want to submit a transaction for a range of [41200 - 61800] Octas at a gas unit price of 100 Octas? [yes/no] >
yes
{
  "Result": {
    "transaction_hash": "0xf04af6810f40b3b6ba6de5445c9c7896c4fad12ef09aa523c3d2c1c7839e7702",
    "gas_used": 412,
    "gas_unit_price": 100,
    "sender": "b11affd5c514bb969e988710ef57813d9556cc1e3fe6dc9aa6a82b56aee53d98",
    "sequence_number": 3,
    "success": true,
    "timestamp_us": 1675409546603551,
    "version": 9800433,
    "vm_status": "Executed successfully"
  }
}

aptos move view --function-id devnet::message::get_message --profile devnet --args address:devnet                   
{
  "Result": [
    "I can use view functions"
  ]
}
```

### CODE REVIEW

File: aptos-move/move-examples/hello_blockchain/sources/hello_blockchain.move [hunk 1]
```diff
@@ -20,6 +20,7 @@ module hello_blockchain::message {
     /// There is no message present
     const ENO_MESSAGE: u64 = 0;
 
+    #[view]
```
Comment (0xmaayan) [inline aptos-move/move-examples/hello_blockchain/sources/hello_blockchain.move:23]: I think we have a tutorial uses this file, do we need to update something there also?

File: crates/aptos/src/move_tool/mod.rs [hunk 2]
```diff
@@ -1396,6 +1452,71 @@ impl ArgWithType {
...
+            FunctionArgType::Bool => serde_json::to_value(bcs::from_bytes::<bool>(&self.arg)?),
+            FunctionArgType::Hex => serde_json::to_value(bcs::from_bytes::<Vec<u8>>(&self.arg)?),
+            FunctionArgType::String => serde_json::to_value(bcs::from_bytes::<String>(&self.arg)?),
+            FunctionArgType::U8 => serde_json::to_value(bcs::from_bytes::<u32>(&self.arg)?),
+            FunctionArgType::U16 => serde_json::to_value(bcs::from_bytes::<u32>(&self.arg)?),
+            FunctionArgType::U32 => serde_json::to_value(bcs::from_bytes::<u32>(&self.arg)?),
+            FunctionArgType::U64 => {
+                serde_json::to_value(bcs::from_bytes::<u64>(&self.arg)?.to_string())
+            },
+            FunctionArgType::U128 => {
```
Comment (0xmaayan) [inline crates/aptos/src/move_tool/mod.rs:1470]: looks like a duplicated code? maybe move it to a function?

**Review by 0xmaayan — APPROVED** [review]

File: crates/aptos/src/move_tool/mod.rs [hunk 3]
```diff
@@ -1101,6 +1105,58 @@ impl CliCommand<TransactionSummary> for RunFunction {
...
+    ///
+    /// Example: `0x842ed41fad9640a2ad08fdd7d3e4f7f505319aac7d67e1c0dd6a7cce8732c7e3::message::set_message`
+    #[clap(long)]
+    pub(crate) function_id: MemberId,
+
+    /// Arguments combined with their type separated by spaces.
+    ///
+    /// Supported types [u8, u16, u32, u64, u128, u256, bool, hex, string, address, raw, vector<inner_type>]
+    ///
+    /// Example: `address:0x1 bool:true u8:0 u256:1234 'vector<u32>:a,b,c,d'`
```
Comment (davidiw) [inline crates/aptos/src/move_tool/mod.rs:1121]: we really need to get to using the same api that rest does

**Review by davidiw — COMMENTED** [review]
also don't we already have this code somewhere for other applications -- like generation tns?

**Review by davidiw — APPROVED** [review]

File: crates/aptos/src/move_tool/mod.rs (hunk 3 above)
Comment (gregnazario) [reply crates/aptos/src/move_tool/mod.rs:1121]: You're saying just taking in JSON strings?  We could do that.  But when it uses BCS, it's harder to do from a command line.

Or are you saying automatic BCS conversion

**Pull request merged**